app/controllers/responses_controller.rb

```diff
@@ -58,25 +58,26 @@ def index
       # csv output is for exporting responses
       format.csv do
         authorize!(:export, Response)
-        @responses = @responses.accessible_by(current_ability, :export)
 
-        # do search, excluding excerpts
-        if params[:search].present?
-          begin
-            @responses = Response.do_search(@responses, params[:search], {mission: current_mission},
-              include_excerpts: false)
-          rescue Search::ParseError => error
-            flash.now[:error] = error.to_s
-            return
-          end
-        end
-
-        # Get the response, for export, but not paginated.
-        # We deliberately don't eager load as that is handled in the Results::Csv::Generator class.
-        @responses = @responses.order(:created_at)
-
-        @csv = Results::Csv::Generator.new(@responses)
-        render_csv("elmo-#{current_mission.compact_name}-responses-#{Time.zone.now.to_s(:filename_datetime)}")
+        operation = Operation.new(
+          creator: current_user,
+          job_class: ResponseCsvExportOperationJob,
+          description: t(
+            "operation.description.response_csv_export_operation_job",
+            user_email: current_user.email,
+            mission_name: current_mission.name
+          )
+        )
+
+        operation.begin!(
+          current_mission,
+          params[:search]
+        )
```

Comment:
One line please.

Reply:
Can we also put the operation creation and begin! in a helper method. This method is super long!